lib/ansible/module_utils/network/nxos/nxos.py

```diff
@@ -969,24 +970,38 @@ def get_proposed(self):
                 else:
                     cmd = ref[k]['setval'].format(playval)
             else:
-                raise "get_proposed: unknown 'kind' value specified for key '{0}'".format(k)
+                raise ValueError("get_proposed: unknown 'kind' value specified for key '{0}'".format(k))
             if cmd:
                 if 'absent' == ref['_state'] and not re.search(r'^no', cmd):
                     cmd = 'no ' + cmd
-                # This command may require parent commands for proper context
-                # Global _template context is replaced by parameter context
-                parent_context = ref['_template'].get('context', [])
-                parent_context = ref[k].get('context', parent_context)
-                for context in parent_context:
-                    if isinstance(context, list):
-                        for ctx_cmd in context:
-                            proposed.append(ctx_cmd)
-                    elif isinstance(context, str):
-                        proposed.append(context)
-
-                proposed.append(cmd)
-
-        return proposed
+                # Add processed command to cmd_ref object
+                ref[k]['setcmd'] = cmd
+
```

Comment:
I'd prefer to have the new code below as a separate method; e.g. `proposed_context()`.